x/concentrated-liquidity/lp_test.go

```diff
@@ -978,15 +967,23 @@ func (s *KeeperTestSuite) TestAddToPosition() {
 			// We expect the position ID to be 3 since we have two setup positions
 			s.Require().Equal(uint64(3), newPosId)
 
-			// Ensure balances were deducted by the correct amounts
-			// Note that we subtract rounding error from the initial amount of
-			// both assets since both are truncated upon withdrawal (so there is at least one
-			// unit of each left in the pool).
-			postSendBalanceSender := s.App.BankKeeper.GetAllBalances(s.Ctx, sender)
-			s.Require().Equal(
-				sdk.NewCoins(sdk.NewCoin(pool.GetToken0(), config.amount0Expected.Sub(amount0Initial.Sub(roundingError))), sdk.NewCoin(pool.GetToken1(), config.amount1Expected.Sub(amount1Initial.Sub(roundingError)))),
-				preSendBalanceSender.Sub(postSendBalanceSender),
-			)
+			expectedAmount1Delta := sdk.ZeroInt()
+
+			// delta amount1 only exists if the actual amount from addToPosition is not equivilent to tokens provided.
+			// delta amount1 is calculated via (amount1 to create initial position) + (amount1 added to position) - (actual amount 1)
+			if fundCoins.AmountOf(pool.GetToken1()).Add(tc.amount1ToAdd).Sub(newAmt1).GT(sdk.ZeroInt()) {
+				expectedAmount1Delta = config.tokensProvided.AmountOf(pool.GetToken1()).Add(tc.amount1ToAdd).Sub(newAmt1)
+			}
+
+			postBalanceToken0 := s.App.BankKeeper.GetBalance(s.Ctx, sender, pool.GetToken0())
+			postBalanceToken1 := s.App.BankKeeper.GetBalance(s.Ctx, sender, pool.GetToken1())
+
+			var errTolerance osmomath.ErrTolerance
```

Comment:
Just to be sure: which rounding error is this meant to accommodate? I noticed in the test cases you added the withdraw amounts are rounded down in the expected values, so I'm assuming this is for the added values?

Reply:
yes this is for the rounding errors caused in the process of withdrawing